index.js

```diff
@@ -117,14 +117,31 @@ module.exports = fp(function (fastify, options, next) {
     }
 
     const session = new Session(JSON.parse(msg))
-    session.changed = signingKeyRotated
+
+    if (session[kObj].state) {
```

Comment:
Why is the state property special?

Reply:
it is actually the problem of `passport-oauth2` which directly mutating the `request.session` object.
The usage of `fastify-secure-session` only allow to use the `get` and `set` function to mutate the `session`.

Reply:
I can only think of adding an extra layer `new Proxy()` to support this use-case.

Reply:
@climba03003 yes, I think you are right. If `passport-oauth2` uses `get` and `set` functions to mutate the session then this problem won't exist